### PR TITLE
Issue #357: Add 'Cancel' link on node forms.

### PR DESCRIPTION
**Pull request by ghost**
Addresses https://github.com/backdrop/backdrop-issues/issues/357.

Adds 'Cancel' link which takes the user to first of:
- URL 'destination' parameter
- Previous page
- Node page
- Frontpage

### CODE REVIEW

**Comment by jenlampton** [comment]
Can we please make these buttons into a link?  
short reason: it needs to be less visually identifiable than save/submit

File: core/modules/node/node.pages.inc [hunk 1]
```diff
@@ -366,6 +382,13 @@ function node_form($form, &$form_state, Node $node) {
       '#submit' => array('node_form_delete_submit'),
     );
   }
+  $form['actions']['cancel'] = array(
+    '#type' => 'link',
```
Comment (ghost) [inline core/modules/node/node.pages.inc:386]: It is already a link...

EDIT: I simply copied the existing Cancel link from the node delete form (and added custom code for choosing which URL to load).

**Comment by quicksketch** [comment]
This looks great to me! Neat trick with the HTTP_REFERER.

**Comment by jenlampton** [comment]
Sorry @BWPanda I saw a form element and assumed it was a button.  I was expecting #type = markup and not link.  Good work ;)

**Pull request merged**